bin/proving-service/src/proxy/mod.rs

```diff
@@ -117,13 +120,18 @@ impl LoadBalancerState {
         })
     }
 
-    /// Marks the given worker as available.
+    /// Marks the given worker as available and moves it to the end of the list.
     ///
     /// If the worker is not in the list, it won't be added.
     pub async fn add_available_worker(&self, worker: Worker) {
```

Comment:
I would add a brief explanation of why we more the worker to the end of the list.